pkg/odo/genericclioptions/context.go

```diff
@@ -74,8 +81,27 @@ func newContext(command *cobra.Command, createAppIfNeeded bool) *Context {
 	} else {
 		// otherwise use the current project
 		ns = project.GetCurrent(client)
+		// if no current project, then check if user is trying to create or delete something other than project
+		if len(ns) <= 0 {
+			errFormat := "could not get current project.  Please create or set a project\n\t%s project create|set <project_name>"
+			checkProjectCreateOrDeleteOnlyOnInvalidNamespace(command, errFormat)
+		}
+		// If 'default' project set then check if user is trying to create or delete something other than project
+		if ns == "default" {
+			errFormat := "Current project is default.  Please create or set a different project\n\t%s project create|set <project_name>"
```

Comment:
There is an extra space after your first sentence.

Reply:
Addressed